### PR TITLE
feat(gherkin-track): Added new steps and support for testing track API.

### PR DESCRIPTION
**Pull request by yasirfolio3**
### Summary
- Added steps to process track API.
- General fixes.

### CODE REVIEW

File: tests/integration/models/get_enabled_features_params.go [hunk 1]
```diff
@@ -16,8 +16,8 @@
 
 package models
 
-// GetEnabledFeaturesParams represents params required for GetEnabledFeatures API
-type GetEnabledFeaturesParams struct {
+// GetEnabledFeaturesRequestParams represents params required for GetEnabledFeatures API
+type GetEnabledFeaturesRequestParams struct {
```
Comment (msohailhussain) [inline tests/integration/models/get_enabled_features_params.go]: why you changed names?

File: tests/integration/main_test.go [hunk 2]
```diff
@@ -69,4 +69,5 @@ func FeatureContext(s *godog.Suite) {
 	s.Step(`^the number of dispatched events is (\d+)$`, context.TheNumberOfDispatchedEventsIs)
 	s.Step(`^there are no dispatched events$`, context.ThereAreNoDispatchedEvents)
 	s.Step(`^dispatched events payloads include$`, context.DispatchedEventsPayloadsInclude)
+	s.Step(`^payloads of dispatched events don\'t include decisions$`, context.PayloadsOfDispatchedEventsDontIncludeDecisions)
```
Comment (msohailhussain) [inline tests/integration/main_test.go]: this is interesting, i just realized this step in steps.js

File: tests/integration/models/get_enabled_features_params.go (hunk 1 above)
Comment (yasirfolio3) [reply tests/integration/models/get_enabled_features_params.go]: to follow the same naming scheme for all request params.

File: tests/integration/support/steps.go [hunk 3]
```diff
@@ -56,6 +56,13 @@ func (c *ScenarioCtx) ListenerIsAdded(numberOfListeners int, ListenerName string
 func (c *ScenarioCtx) IsCalledWithArguments(apiName string, arguments *gherkin.DocString) error {
 	c.apiOptions.APIName = apiName
 	c.apiOptions.Arguments = arguments.Content
+
+	// Clearing old state of response, eventdispatcher and decision service
+	c.apiResponse = models.APIResponse{}
+	// Only required for unsessioned tests
+	c.clientWrapper.DecisionService.(*optlyplugins.TestCompositeService).ClearListenersCalled()
```
Comment (mikeng13) [inline tests/integration/support/steps.go]: HOw come we need to clean up between calls? Can we just generate a new client wrapper or "get" a new client wrapper based on the requestID?

Comment (yasirfolio3) [reply tests/integration/support/steps.go]: This is for scenarios where multiple api's are called one after another. In such cases, listeners_called, dispatched_events and response of former request needs to be cleared before the latter request is executed. We only recreate client-wrapper when step "the datafile is" is called so for the given case, same client-wrapper is used.

Comment (mikeng13) [reply tests/integration/support/steps.go]: What I was saying is that we can use the `request-id` from the request params to cached/fetch instances of client wrappers and create the clientWrapper here when we call `IsCalledWithArguments`. That way we don't need to explicitly clean up these components. We already do this in other SDKs for multiple API calls

Comment (yasirfolio3) [reply tests/integration/support/steps.go]: done, can you please review it now.

File: tests/integration/optlyplugins/proxy_dispatcher.go [hunk 4]
```diff
@@ -43,3 +43,8 @@ func (d *ProxyEventDispatcher) GetEvents() []event.Batch {
 	}
 	return d.events
 }
+
+// ClearEvents deletes dispatched events
+func (d *ProxyEventDispatcher) ClearEvents() {
```
Comment (mikeng13) [inline tests/integration/optlyplugins/proxy_dispatcher.go]: This is not used anymore

File: tests/integration/optlyplugins/test_composite_service.go [hunk 5]
```diff
@@ -52,6 +52,11 @@ func (c *TestCompositeService) GetListenersCalled() []models.DecisionListener {
 	return c.listenersCalled
 }
 
+// ClearListenersCalled - Deletes listeners called
+func (c *TestCompositeService) ClearListenersCalled() {
```
Comment (mikeng13) [inline tests/integration/optlyplugins/test_composite_service.go]: This is no longer needed

Comment (yasirfolio3) [reply tests/integration/optlyplugins/test_composite_service.go]: both changes made.

**Pull request merged**